staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3125,6 +3125,22 @@ type HostAlias struct {
 	Hostnames []string `json:"hostnames,omitempty" protobuf:"bytes,2,rep,name=hostnames"`
 }
 
+// PodFSGroupChangePolicy holds policies that will be used for applying fsGroup to a volume
+// when volume is mounted.
+type PodFSGroupChangePolicy string
+
+const (
+	// OnRootMismatch indicates that volume's ownership and permissions will be changed
+	// only when permission and ownership of root directory does not match with expected
+	// permissions on the volume. This can help shorten the time it takes to change
+	// ownership and permissions of a volume.
+	OnRootMismatch PodFSGroupChangePolicy = "OnRootMismatch"
+	// AlwaysChangeVolumePermission indicates that volume's ownership and permissions
+	// should always be changed whenever volume is mounted inside a Pod. This the default
+	// behavior.
+	AlwaysChangeVolumePermission PodFSGroupChangePolicy = "Always"
```

Comment:
Do we want to add defaulting logic?

Reply:
I don't particularly care about defaulting in this case. Do you think it might be beneficial in some ways? Can we add this in next iteration?

Reply:
Since we are documenting the nil behavior as defaulted to one of the values, I think we need to make that change explicit in the API.